Source/AwsCommonRuntimeKit/mqtt/Mqtt5Client.swift

```diff
@@ -107,7 +106,38 @@ public class Mqtt5Client {
                     return continuation.resume(throwing: CommonRunTimeError.crtError(CRTError.makeFromLastError()))
                 }
             }
+        }
+    }
+
+    public func unsubscribe(unsubscribePacket: UnsubscribePacket) async throws -> UnsubackPacket {
+
+        return try await withCheckedThrowingContinuation { continuation in
 
+            func unsubscribeCompletionCallback(unsubackPacket: UnsafePointer<aws_mqtt5_packet_unsuback_view>?,
+                                               errorCode: Int32,
+                                               userData: UnsafeMutableRawPointer?) {
+                let continuationCore = Unmanaged<ContinuationCore<UnsubackPacket>>.fromOpaque(userData!).takeRetainedValue()
+
+                guard errorCode == AWS_OP_SUCCESS else {
+                    return continuationCore.continuation.resume(throwing: CommonRunTimeError.crtError(CRTError(code: errorCode)))
+                }
+
+                guard let unsuback = UnsubackPacket.convertFromNative(unsubackPacket) else {
+                    fatalError("Unsuback missing in the Unsubscribe completion callback.")
+                }
+
+                continuationCore.continuation.resume(returning: unsuback)
+            }
+
+            unsubscribePacket.withCPointer { unsubscribePacketPointer in
+                var callbackOptions = aws_mqtt5_unsubscribe_completion_options()
+                callbackOptions.completion_callback = unsubscribeCompletionCallback
+                callbackOptions.completion_user_data = ContinuationCore(continuation: continuation).passRetained()
+                let result = aws_mqtt5_client_unsubscribe(rawValue, unsubscribePacketPointer, &callbackOptions)
+                guard result == AWS_OP_SUCCESS else {
+                    return continuation.resume(throwing: CommonRunTimeError.crtError(CRTError.makeFromLastError()))
```

Comment:
Just realize if the operation failed, we should release the continuationCore as its reference was acquired by `passRetained()`. I updated the subscribe and the publish operation to release it: 
https://github.com/awslabs/aws-crt-swift/blob/27cd454c3a72ecaaeed4513da19a1a2be18fad03/Source/AwsCommonRuntimeKit/mqtt/Mqtt5Client.swift#L104